### PR TITLE
mongodb: findOne may return null

### PR DESCRIPTION
**Pull request by despairblue**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.mongodb.com/manual/reference/method/db.collection.findOne/#db.collection.findOne (linking to the mongdb shell documenation since the documentation for the native driver does not specify explicitly what it returns if it does not find anything matching the query)
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by despairblue** [comment]
CI fails for mongoose-simple-random:

> ERROR: 6:16  no-declare-current-package  Instead of declaring a module with `declare module "mongoose-simple-random"`, write its contents in directly in "index.d.ts". See: https://github.com/Microsoft/dtslint/blob/master/docs/no-declare-current-package.md

File: types/mongodb/index.d.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export class MongoClient {
 }
 
 export interface MongoCallback<T> {
-    (error: MongoError, result: T): void;
+    (error: MongoError, result: T | null): void;
```
Comment (CaselIT) [inline types/mongodb/index.d.ts]: `MongoCallback` is used in many other signatures. `result` should remain of type `T`. Change the signature of the `findOne` function to `MongoCallback<T|null>` instead.

**Review by CaselIT — CHANGES_REQUESTED** [review]

File: types/mongodb/index.d.ts (hunk 1 above)
Comment (despairblue) [reply types/mongodb/index.d.ts]: done :+1:

**Review by CaselIT — APPROVED** [review]

**Comment by CaselIT** [comment]
👍

**Pull request merged**